### PR TITLE
Add session idle timeout option for session processor

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/16773

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public async Task ProcessMessages(int numThreads, bool autoComplete)
                 enablePartitioning: false,
                 enableSession: false))
             {
-                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+                await using var client = GetClient();
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs:27]: nit:  This looks more like you're creating and disposing a client.   Maybe `CreateClient` would be a more accurate factory method name?

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessorOptions.cs [hunk 2]
```diff
@@ -63,31 +63,31 @@ public TimeSpan MaxAutoLockRenewalDuration
         private TimeSpan _maxAutoRenewDuration = TimeSpan.FromMinutes(5);
 
         /// <summary>
-        /// Gets or sets the maximum amount of time to wait for each Receive call using the processor's underlying
-        /// receiver.
+        /// Gets or sets the maximum amount of time to wait before switching to a new session
+        /// after no messages have been received on the session that is currently being processed.
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessorOptions.cs:67]: Thanks, I will take another stab at it based on your feedback.

**Pull request merged**